### PR TITLE
Fix some compiler warnings

### LINKED ISSUE #333: Field propertyValueExpander in PropertiesConfigSource defined but never used
https://github.com/eclipse/lsp4mp/blob/7591749ca1eb403a72fab7b53f08930b3cf15267/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/PropertiesConfigSource.java#L38

### PR DESCRIPTION
**Pull request by datho7561**
Fixes #333

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

**Comment by datho7561** [comment]
I'm going to clean up a few other simple warnings in this PR

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/PropertiesManager.java [hunk 1]
```diff
@@ -386,11 +386,11 @@ private SearchPattern createSearchPattern() {
 	 */
 	private static IJavaSearchScope createJavaSearchScope(IJavaProject fakeProject, boolean excludeTestCode,
 			IJavaElement[] elements, int includeMask) {
-		HashSet projectsToBeAdded = new HashSet(2);
+		HashSet<JavaProject> projectsToBeAdded = new HashSet<JavaProject>(2);
```
Comment (angelozerr) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/PropertiesManager.java:389]: Set<JavaProject>

Comment (datho7561) [reply microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/PropertiesManager.java:389]: Unfortunately, it looks like a JDT API requires it to be a HashSet, although I'll update it to use the diamond operator.

**Comment by angelozerr** [comment]
Thanks @datho7561 !

**Pull request merged**